### PR TITLE
Bugfix - handle connection failure behavior

### PR DESCRIPTION
**Pull request by tsimchev**
- do not update cache on connection failure
- add additional metric about last successful scrap of NSX-T API
  - this metric will be updated only when the collection from all of the
  NSX-T API endpoints is successful.

### CODE REVIEW

**Comment by joker-official** [comment]
> do not update cache on connection failure
Does that mean we report stale metrics? And the only way to figure this out is by finding the additional metric about the last successful scrape?

**Comment by tsimchev** [comment]
> > do not update cache on connection failure
> 
> Does that mean we report stale metrics? And the only way to figure this out is by finding the additional metric about the last successful scrape?

yes, this is the approach we discussed yesterday.

Some Additional Options that we discussed
- add additional value -999 to metrics, indicating conn failure or time out
- hide metric from the list
- add an additional metric for every metric ending with _since in seconds

**Comment by joker-official** [comment]
OK, if that was the consent.

File: exporter/gather.go [hunk 1]
```diff
@@ -228,15 +262,15 @@ func logicalSwitchStateHander(resp *Nsxv3Response, data *Nsxv3Data) string {
...
 func init() {
-	endpoints = map[string]func(resp *Nsxv3Response, data *Nsxv3Data) string{
+	endpoints = map[string]func(resp *Nsxv3Response, data *Nsxv3Data) (string, error){
 		"/api/v1/cluster/status":         clusterStatusHandler,
-		"/api/v1/cluster/nodes/status":   clusterNodesStatusHandler,
-		"/api/v1/logical-switches":       logicalSwitchAdminStateHander,
-		"/api/v1/logical-switches/state": logicalSwitchStateHander,
+		"/api/v1/cluster/nodes/statufs":   clusterNodesStatusHandler,
+		"/api/v1/logical-switchefs":       logicalSwitchAdminStateHander,
+		"/api/v1/logical-switches/sftate": logicalSwitchStateHander,
```
Comment (notandy) [inline exporter/gather.go]: Is this intended?

Comment (tsimchev) [reply exporter/gather.go]: No, thanks for catching it, these were some of the test endpoints.

**Pull request merged**